src/Commands/Images/GenerateAbstract.php

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * 2019-present Friends of Presta community
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the MIT License
+ * that is bundled with this package in the file LICENSE
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/MIT
+ *
+ * @author Friends of Presta community
+ * @copyright 2019-present Friends of Presta community
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace FOP\Console\Commands\Images;
+
+use Configuration;
+use FOP\Console\Command;
+use Image;
+use ImageManager;
+use ImageType;
+use Language;
+use Module;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+abstract class GenerateAbstract extends Command
+{
+    /** @var array */
+    protected $errors = [];
+
+    /** @var OutputInterface */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+
+        return parent::initialize($input, $output);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:images:generate:' . static::IMAGE_TYPE)
+            ->setDescription('Regenerate ' . static::IMAGE_TYPE . ' thumbnails')
+            ->addArgument(
+                'format',
+                InputArgument::OPTIONAL,
+                'images formats comma separated',
+                'all'
+            )
+            ->addOption(
+                'delete',
+                'd',
+                InputOption::VALUE_OPTIONAL,
+                'Delete current thumbnails',
+                true
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $formats = explode(',', $input->getArgument('format'));
+        $delete = (bool) $input->getOption('delete');
+        $io = new SymfonyStyle($input, $output);
+
+        if (true !== $this->_regenerateThumbnails(static::IMAGE_TYPE, $delete, $formats)) {
+            $io->error('Unable to generate thumbnails');
+
+            return 1;
+        }
+
+        if (count($this->errors)) {
+            $io->error('The generation generate the folowing errors');
+            foreach ($this->errors as $error) {
+                $io->error($error);
+            }
+        }
+
+        $io->success('Thumbnails generated with success for ' . static::IMAGE_TYPE);
```

Comment:
We should display number or regenerated images here.

Reply:
A Success + warning may be better..

Reply:
The last modif of @SebSept add the return so the success message is not displayed anymore in case of error.
I've just improved the display of the error to use just one block instead of one block for each errors.
![image](https://user-images.githubusercontent.com/7163132/74334359-0c9adf00-4d9a-11ea-8a18-803d4994e862.png)
Good for everyone now ?

Reply:
I'll test later, today.

Reply:
ping @SebSept  possible to review to finish the first iteration of this command :) ?

Reply:
There is one block but it still huge. too long for output.
Whatever, let's validate it, we'll improve later.